### ISSUE TITLE
[solve]Exception: path is on mount 'C:', start on mount 'D:'. Cache may be out of date,

### ISSUE THREAD

**Issue by polarbear0827** [ultralytics/yolov5#6976]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Detection

### Bug

D:\Users\a5201\yolococogpu01\venv\Scripts\python.exe D:/Users/a5201/yolococogpu01/detect01.py
Downloading: "https://github.com/ultralytics/yolov5/archive/master.zip" to C:\Users\USER/.cache\torch\hub\master.zip
fatal: not a git repository (or any of the parent directories): .git
YOLOv5  2022-3-14 torch 1.8.2+cu111 CUDA:0 (NVIDIA GeForce GTX 950M, 2048MiB)

Traceback (most recent call last):
  File "C:\Users\USER/.cache\torch\hub\ultralytics_yolov5_master\hubconf.py", line 46, in _create
    model = DetectMultiBackend(path, device=device)  # download/load FP32 model
  File "C:\Users\USER/.cache\torch\hub\ultralytics_yolov5_master\models\common.py", line 297, in __init__
    pt, jit, onnx, xml, engine, coreml, saved_model, pb, tflite, edgetpu, tfjs = self.model_type(w)  # get backend
  File "C:\Users\USER/.cache\torch\hub\ultralytics_yolov5_master\models\common.py", line 475, in model_type
    from export import export_formats
  File "C:\Users\USER/.cache\torch\hub\ultralytics_yolov5_master\export.py", line 64, in <module>
    ROOT = Path(os.path.relpath(ROOT, Path.cwd()))  # relative
  File "C:\Users\USER\AppData\Local\Programs\Python\Python39\lib\ntpath.py", line 703, in relpath
    raise ValueError("path is on mount %r, start on mount %r" % (
ValueError: path is on mount 'C:', start on mount 'D:'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\Users\a5201\yolococogpu01\detect01.py", line 5, in <module>
    model = torch.hub.load("ultralytics/yolov5", "custom", path = "yolov5/runs/train/exp34/weights/best.pt", force_reload=True ).autoshape()
  File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 339, in load
    model = _load_local(repo_or_dir, model, *args, **kwargs)
  File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 368, in _load_local
    model = entry(*args, **kwargs)
  File "C:\Users\USER/.cache\torch\hub\ultralytics_yolov5_master\hubconf.py", line 70, in custom
    return _create(path, autoshape=autoshape, verbose=verbose, device=device)
  File "C:\Users\USER/.cache\torch\hub\ultralytics_yolov5_master\hubconf.py", line 65, in _create
    raise Exception(s) from e
Exception: path is on mount 'C:', start on mount 'D:'. Cache may be out of date, try `force_reload=True` or see https://github.com/ultralytics/yolov5/issues/36 for help.

Process finished with exit code 1

### Environment

YOLOv5  2022-3-14 torch 1.8.2+cu111 CUDA:0 (NVIDIA GeForce GTX 950M, 2048MiB)
Python3.9

### Additional

code:
import torch
import cv2
import numpy as np

model = torch.hub.load("ultralytics/yolov5", "custom", path = "yolov5/runs/train/exp34/weights/best.pt", force_reload=True )

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by quangdungluong** [comment]
I have the same problem as you. The possible reason for this error is the `os.relpath` does not give you a relative path between two directories. I tried load model by clone the repo and load local and it works.
` model = torch.hub.load(path/to/yolov5, 'custom', path=path/to/your/model ,source='local')`
You also avoid this problem by copying your app to C: drive, selecting your python interpreter and trying to execute the code again.

**Comment by polarbear0827** [comment]
Is it changed to this?

model = torch.hub.load("C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master", "custom", path = "D:/Users/a5201/olococogpu01/yolov5/runs/train/exp34/weights/best.pt", force_reload=True )

**Comment by polarbear0827** [comment]
it produces another error

D:\Users\a5201\yolococogpu01\venv\Scripts\python.exe D:/Users/a5201/yolococogpu01/detect01.py
Traceback (most recent call last):
  File "D:\Users\a5201\yolococogpu01\detect01.py", line 5, in <module>
    model = torch.hub.load("C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master", "custom", path = "D:/Users/a5201/olococogpu01/yolov5/runs/train/exp34/weights/best.pt", force_reload=True )
  File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 337, in load
    repo_or_dir = _get_cache_or_reload(repo_or_dir, force_reload, verbose)
  File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 121, in _get_cache_or_reload
    repo_owner, repo_name, branch = _parse_repo_info(github)
  File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 111, in _parse_repo_info
    repo_owner, repo_name = repo_info.split('/')
ValueError: not enough values to unpack (expected 2, got 1)

**Comment by quangdungluong** [comment]
> Is it changed to this?
> 
> 
> 
> model = torch.hub.load("C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master", "custom", path = "D:/Users/a5201/olococogpu01/yolov5/runs/train/exp34/weights/best.pt", force_reload=True )
> 
> 

I haven't tried it. I cloned yolov5 repo and it works.

**Comment by quangdungluong** [comment]
> it produces another error
> 
> 
> 
> D:\Users\a5201\yolococogpu01\venv\Scripts\python.exe D:/Users/a5201/yolococogpu01/detect01.py
> 
> Traceback (most recent call last):
> 
>   File "D:\Users\a5201\yolococogpu01\detect01.py", line 5, in <module>
> 
>     model = torch.hub.load("C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master", "custom", path = "D:/Users/a5201/olococogpu01/yolov5/runs/train/exp34/weights/best.pt", force_reload=True )
> 
>   File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 337, in load
> 
>     repo_or_dir = _get_cache_or_reload(repo_or_dir, force_reload, verbose)
> 
>   File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 121, in _get_cache_or_reload
> 
>     repo_owner, repo_name, branch = _parse_repo_info(github)
> 
>   File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 111, in _parse_repo_info
> 
>     repo_owner, repo_name = repo_info.split('/')
> 
> ValueError: not enough values to unpack (expected 2, got 1)
> 
> 

remember "source=local"

**Comment by polarbear0827** [comment]
i'd like to apologize for my forgot that, but it still error

model = torch.hub.load("C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master", "custom", path = "D:/Users/a5201/olococogpu01/yolov5/runs/train/exp34/weights/best.pt", source='local' )

fatal: not a git repository (or any of the parent directories): .git
YOLOv5  2022-3-14 torch 1.8.2+cu111 CUDA:0 (NVIDIA GeForce GTX 950M, 2048MiB)

Traceback (most recent call last):
  File "C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master\hubconf.py", line 46, in _create
    model = DetectMultiBackend(path, device=device)  # download/load FP32 model
  File "C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master\models\common.py", line 297, in __init__
    pt, jit, onnx, xml, engine, coreml, saved_model, pb, tflite, edgetpu, tfjs = self.model_type(w)  # get backend
  File "C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master\models\common.py", line 476, in model_type
    from export import export_formats
  File "C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master\export.py", line 64, in <module>
    ROOT = Path(os.path.relpath(ROOT, Path.cwd()))  # relative
  File "C:\Users\USER\AppData\Local\Programs\Python\Python39\lib\ntpath.py", line 703, in relpath
    raise ValueError("path is on mount %r, start on mount %r" % (
ValueError: path is on mount 'C:', start on mount 'D:'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\Users\a5201\yolococogpu01\detect01.py", line 5, in <module>
    model = torch.hub.load("C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master", "custom", path = "D:/Users/a5201/olococogpu01/yolov5/runs/train/exp34/weights/best.pt", source='local' )
  File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 339, in load
    model = _load_local(repo_or_dir, model, *args, **kwargs)
  File "D:\Users\a5201\yolococogpu01\venv\lib\site-packages\torch\hub.py", line 368, in _load_local
    model = entry(*args, **kwargs)
  File "C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master\hubconf.py", line 70, in custom
    return _create(path, autoshape=autoshape, verbose=verbose, device=device)
  File "C:/Users/USER/.cache/torch/hub/ultralytics_yolov5_master\hubconf.py", line 65, in _create
    raise Exception(s) from e
Exception: path is on mount 'C:', start on mount 'D:'. Cache may be out of date, try `force_reload=True` or see https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading for help.

Process finished with exit code 1

**Comment by quangdungluong** [comment]
Lets try clone yolov5 repo and copy path. It works for me

**Comment by polarbear0827** [comment]
I clone yolov5 and the problem still occurs

**Comment by polarbear0827** [comment]
Move entire file to C folder， It can be solved

**Comment by wonderfuer** [comment]
In general, allocate a directory on Disk C to be used as the running directory of the API site, and then copy restapi.py to solve the problem.

**Comment by glenn-jocher** [comment]
@polarbear0827 @quangdungluong @chenzhikang I have a proposed fix PR implemented in #7158. Can you please test the PR with this command? If it works I'll merge into master, if it doesn't work please paste your new error message. Thanks!

```python
import torch

model = torch.hub.load('ultralytics/yolov5:fix/relpath', 'yolov5s', force_reload=True)
```

**Comment by glenn-jocher** [comment]
@polarbear0827 @quangdungluong @chenzhikang good news 😃! Your original issue may now be fixed ✅ in PR #7158. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
